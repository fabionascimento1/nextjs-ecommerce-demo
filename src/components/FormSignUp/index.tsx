import { useState } from 'react'

import Link from 'next/link'
import { useRouter } from 'next/router'

import {
  AccountCircle,
  Email,
  Lock,
  ErrorOutline
} from '@styled-icons/material-outlined'

import { FormLink, FormWrapper, FormLoading, FormError } from 'components/Form'
import Button from 'components/Button'
import TextField from 'components/TextField'

import * as S from './styles'

const FormSignUp = () => {
  return (
    <FormWrapper>
      <form>
        <TextField
          name="name"
          placeholder="Name"
          type="text"
          error=""
          icon={<AccountCircle />}
        />
        <TextField
          name="email"
          placeholder="Email"
          type="email"
          error=""
          icon={<Email />}
        />
        <TextField
          name="password"
          placeholder="Password"
          type="password"
          error=""
          icon={<Lock />}
        />
        <TextField
          name="confirm-password"
          placeholder="Confirm Password"
          type="password"
          error=""
          icon={<Lock />}
        />

        <Button type="submit" size="large" fullWidth>
          sign up now
        </Button>

        <FormLink>
          Você já possui uma conta?{' '}
          <Link href="/signin">
            <a>Sign in</a>
          </Link>
        </FormLink>
      </form>
    </FormWrapper>
  )
}

export default FormSignUp
