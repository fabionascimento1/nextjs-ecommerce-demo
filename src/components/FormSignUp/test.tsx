import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import FormSignUp from '.'

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const push = jest.fn()

useRouter.mockImplementation(() => ({
  push,
  query: '',
  asPath: '',
  route: '/'
}))

describe('<FormSignUp />', () => {
  it('should render the form FormSignUp', () => {
    const { container } = renderWithTheme(<FormSignUp />)

    expect(screen.getByPlaceholderText(/name/i)).toBeInTheDocument()
    expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Password')).toBeInTheDocument()
    expect(screen.getByPlaceholderText(/confirm password/i)).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /sign up now/i })
    ).toBeInTheDocument()

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should render text to sign in if have an account', () => {
    renderWithTheme(<FormSignUp />)

    expect(screen.getByRole('link', { name: /sign in/i })).toBeInTheDocument()
    expect(screen.getByText(/possui uma conta\?/i)).toBeInTheDocument()
  })
})
