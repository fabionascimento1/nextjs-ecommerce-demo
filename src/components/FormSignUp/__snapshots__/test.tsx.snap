// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignUp /> should render the form FormSignUp 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #CECECE;
  border-radius: 0.2rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #CECECE;
}

.c2:focus-within {
  box-shadow: 0 0 0.5rem #460099;
}

.c5 {
  color: #16193A;
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem 0;
  padding-left: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c5:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4rem #CECECE inset;
  -webkit-filter: none;
  filter: none;
}

.c5:-webkit-autofill::first-line {
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #B5B5B5;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c3 > svg {
  width: 2.2rem;
  height: 100%;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#6dfdfc 0%,#1ed6ff 100%);
  color: #460099;
  border-radius: 0.4rem;
  padding: 0.8rem;
  font-weight: 500;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  width: 100%;
}

.c7:hover {
  background: linear-gradient(180deg,#1ed6ff 0%,#6dfdfc 100%);
}

.c0 .c1 {
  margin: 0.8rem 0;
}

.c0 .c6 {
  margin: 3.2rem auto 1.6rem;
}

.c8 {
  font-size: 1.4rem;
  color: #16193A;
  text-align: center;
}

.c8 a {
  color: #6dfdfc;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.1rem solid #6dfdfc;
  -webkit-transition: color,border,0.1s ease-in-out;
  transition: color,border,0.1s ease-in-out;
}

.c8 a:hover {
  color: #3bfcfb;
  border-bottom: 0.1rem solid #3bfcfb;
}

<body>
  <div>
    <div
      class="c0"
    >
      <form>
        <div
          class="c1 "
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <svg
                aria-hidden="true"
                class="c4"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33A7.95 7.95 0 014 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
                />
              </svg>
            </div>
            <input
              class="c5"
              name="name"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 "
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <svg
                aria-hidden="true"
                class="c4"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
                />
              </svg>
            </div>
            <input
              class="c5"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 "
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <svg
                aria-hidden="true"
                class="c4"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    opacity="0.87"
                  />
                </g>
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                />
              </svg>
            </div>
            <input
              class="c5"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 "
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <svg
                aria-hidden="true"
                class="c4"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    opacity="0.87"
                  />
                </g>
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                />
              </svg>
            </div>
            <input
              class="c5"
              name="confirm-password"
              placeholder="Confirm Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          class="c6 c7"
          type="submit"
        >
          <span>
            sign up now
          </span>
        </button>
        <div
          class="c8"
        >
          Você já possui uma conta?
           
          <a
            href="/signin"
          >
            Sign in
          </a>
        </div>
      </form>
    </div>
  </div>
</body>
`;
